prometheus/src/main/scala/filodb/prometheus/ast/Vectors.scala

```diff
@@ -87,6 +93,11 @@ trait Vectors extends Scalars with TimeUnits with Base {
           case NotEqual(false) => ColumnFilter(labelMatch.label, query.Filter.NotEquals(labelMatch.value))
           case other: Any      => throw new IllegalArgumentException(s"Unknown match operator $other")
         }
+      }.filterNot(_.column == ColumnSelectorLabel)
```

Comment:
Can we document here why we are removing this column from the filters?

From my understanding, it is because the selected column names are part of the logical plan as a different input argument.